coreos-installer

```diff
@@ -378,6 +378,42 @@ write_networking_opts() {
     echo "set ignition_network_kcmdline=\"$(cat /tmp/networking_opts)\"" >> /mnt/boot_partition/ignition.firstboot
 }
 
+#########################################################
+# Helper to persist ignition platform id to initramfs first boot
+#########################################################
+write_platform_id() {
+    local tmp_platform_id="/tmp/platform_id"
+
+    if [ ! -f ${tmp_platform_id} ]; then
+        log "Not overwriting ignition platform id, no platform id provided"
+        return
+    else
+        local platform_id="$(cat ${tmp_platform_id})"
+    fi
+
+
+    log "Overwriting ignition platform id"
+    # check for the boot partition
+    mkdir -p /mnt/boot_partition
+    mount_boot_partition /mnt/boot_partition
+    trap 'umount /mnt/boot_partition; trap - RETURN' RETURN
+    local conf_files=$(ls /mnt/boot_partition/loader/entries/*conf)
+
+    for f in $conf_files
+    do
+        if grep -q -E "ignition.platform.id=[^$ ]*" $f; then
+            # ignition.platform.id is followed by whitespace or EOF
+            sed -i -r "s/ignition.platform.id=[^$ ]*?/ignition.platform.id=${platform_id}/" $f
```

Comment:
`$` has no special meaning inside brackets.

Reply:
Updated!

Reply:
that was my fault. I had just tested with `[^$ ]` and it worked. Thanks for the correction @bgilbert